frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/use-join.ts

```diff
@@ -46,10 +63,17 @@ export function useJoin(query: Lib.Query, stageIndex: number, join?: Lib.Join) {
     if (!previousJoin && join) {
       _setSelectedColumns([]);
     }
-  }, [query, previousQuery, stageIndex, join, previousJoin]);
+  }, [
+    query,
+    previousQuery,
+    stageIndex,
+    join,
+    previousJoin,
+    isMarkedForRemoval,
+  ]);
 
   const isColumnSelected = (column: Lib.ColumnMetadata) => {
-    if (join) {
+    if (join && !isMarkedForRemoval) {
```

Comment:
Perhaps it would make sense to do something like:

```ts
function useJoin(inputJoin: Lib.Join) {
  const join = isMarkedForRemoval ? undefined : inputJoin
}
```

to make these checks unnecessary